examples/MLIRPython/buddy/compiler.py

```diff
@@ -72,15 +72,22 @@ def import_graph(self) -> ir.Module:
     """Import the FX graph, generate an MLIR module in high-level dialects.
 
     Returns:
-      mlir.ir.Module: An MLIR moduel in high-level dialects.
+      mlir.ir.Module: An MLIR module in high-level dialects.
 
     """
     with ir.InsertionPoint(self._module.body):
       arguments = []
       for arg in self._inputs:
         shape_list = list(arg.shape)
-        f32 = ir.F32Type.get()
-        tensor_arg = ir.RankedTensorType.get(shape_list, f32)
+        dtype = arg.dtype
+        if dtype is torch.int32:
+          mlir_dtype = ir.IntegerType.get_signless(32)
+        elif dtype is torch.float32:
+          mlir_dtype = ir.F32Type.get()
+        else:
```

Comment:
I think it would be more elegant to use a statement like switch here, since more types may be supported later, and I think you can refer to the match statement.